### ISSUE TITLE
Ignore generated methods for Memoized annotation for plugin validation

### ISSUE THREAD

**Issue by wolfs** [gradle/gradle#16117]
Our current work validation (`ValidatePlugins` task) finds a problem with [this method in the Gradle docker plugin](https://github.com/bmuschko/gradle-docker-plugin/blob/6c7b747a31210faae07f6ad2bbbdde2b95a96d9a/src/main/groovy/com/bmuschko/gradle/docker/tasks/AbstractDockerRemoteApiTask.groovy#L142-L145):
```
    @Internal
    @Memoized
    DockerClient getDockerClient() {
        ...
    }
```

Namely, it reports `non-property method 'memoizedMethodPriv\$getDockerClient()' should not be annotated with: @Internal.`

We should ignore these methods generated by Groovy either altogether or the method generated by the `@Memoized` annotation in particular.

---
cc: @gradle/execution

**Comment by wolfs** [comment]
The change has not been included in Kotlin 1.4.30. The PR https://github.com/jetbrains/kotlin/pull/4064 has been approved, but not merged and now has conflicts :(.

**Comment by melix** [comment]
@wolfs did you mean to comment on this ticket? It seems unrelated.
